File: server/matchmaking/src/me/eddiep/ghost/matchmaking/network/packets/UpdateSessionPacket.java
package me.eddiep.ghost.matchmaking.network.packets;

import me.eddiep.ghost.matchmaking.network.PlayerClient;
import me.eddiep.ghost.matchmaking.network.TcpClient;
import me.eddiep.ghost.matchmaking.network.TcpServer;
import me.eddiep.ghost.network.packet.Packet;

import java.io.IOException;

public class UpdateSessionPacket extends Packet<TcpServer, PlayerClient> {
    public UpdateSessionPacket(PlayerClient client) {
        super(client);
    }

    @Override
    public void onWritePacket(PlayerClient client, Object... args) throws IOException {

        write((byte) 0x29)
                .write(client.getPlayer().getSession().length())
                .write(client.getPlayer().getSession())
                .endTCP();
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/UpdateSessionPacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.Ghost
import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient


class UpdateSessionPacket : Packet<PlayerClient>(){
    override fun handle(){
        val newSession = consume(consume(4).asInt()).asString()

        Ghost.Session = newSession
    }
}
